main.go

```diff
@@ -517,10 +518,12 @@ func finishUpload(num int) error {
 		return err
 	}
 
-	err = os.Rename(path.Join(directory, "tmp", strconv.Itoa(num)+".asc"),
-		path.Join(directory, up.Image+".asc"))
-	if err != nil {
-		return err
+	tmpsig := path.Join(directory, "tmp", strconv.Itoa(num) + ".asc")
+	if _, err := os.Stat(tmpsig); err == nil {
+		err = os.Rename(tmpsig, path.Join(directory, up.Image+".asc"))
+		if err != nil {
+			return err
+		}
```

Comment:
its required else the upload will complete with the Rename error.

I think checking if the file exists is safer than looking for the flag. But I can change if you want.